lib/wallet/src/Cardano/Wallet/Shelley/Transaction.hs

```diff
@@ -711,11 +713,14 @@ mkUnsignedTransaction networkId stakeCred ctx selection = do
                         inpsScripts
                     Nothing refScriptM
         Just action -> do
+            let deposit = view #txDeposit ctx
             let certs = case stakeCred of
```

Comment:
done. I do think `nub $ delegCerts <> votingCerts` should be enough as we use the same staking key during registration

Reply:
> I do think nub $ delegCerts <> votingCerts should be enough as we use the same staking key during registration

Ah, `nub` is a good idea.

One small hitch: We have to careful whether the certificates are applied left-to-right, or right-to-left. In the situation

```hs
  nub ["Delegate", "Register", "Vote", "Register"]
=
  nub ["Delegate", "Register", "Vote"] 
```

applying right-to-left would be a mistake, because `nub` takes the leftmost element when deduplicating.